gadget/gadget.go

```diff
@@ -1385,3 +1385,33 @@ func parseCommandLineFromGadget(content []byte) (string, error) {
 	}
 	return strings.Join(kargs, " "), nil
 }
+
+// HasRole reads the gadget specific metadata from meta/gadget.yaml in the snap
+// root directory with minimal validation and checks whether any volume
+// structure has one of the given roles returning it, otherwhise it returns the
+// empty string.
+func HasRole(gadgetSnapRootDir string, roles []string) (foundRole string, err error) {
```

Comment:
Should this be used just by snap-bootstrap because of the minimal validation? A clarifying comment would be nice to have.